src/WTG.Analyzers.Test/AnalyzerAndCodeFixTest.cs

```diff
@@ -75,17 +77,62 @@ public async Task RunTest([ValueSource(nameof(Samples))] SampleDataSet data)
 
 			Assert.That(actual.Where(filter), IsDiagnostic.EqualTo(data.Diagnostics));
 
-			var fixer = new CodeFixer(analyzer, new TCodeFix())
+			if ((data.Options & SampleDataSetOptions.AllowCodeFixes) != 0)
 			{
-				DiagnosticFilter = filter,
-			};
+				var fixer = new CodeFixer(analyzer, new TCodeFix())
+				{
+					DiagnosticFilter = filter,
+				};
+
+				await fixer.VerifyFixAsync(document, data.Result).ConfigureAwait(false);
+			}
+			else
+			{
+				var codeFix = new TCodeFix();
+				var actions = new List<Tuple<Diagnostic, CodeAction>>();
+
+				foreach (Diagnostic diagnostic in actual)
+				{
+					if (!codeFix.FixableDiagnosticIds.Contains(diagnostic.Id))
+					{
+						continue;
+					}
+
+					await CodeFixUtils.CollectCodeActions(codeFix, document, diagnostic, actions).ConfigureAwait(false);
+				}
+
+				if (actions.Count > 0)
+				{
+					var builder = new StringBuilder()
+						.AppendLine("The diagnostics should not have offered any code fix actions, but instead found:");
+
+					foreach (var (diagnostic, action) in actions)
+					{
+						builder
+							.Append("* ")
+							.Append(diagnostic.Id)
+							.Append(' ');
+
+						AppendLocation(builder, diagnostic.Location);
+
+						builder
+							.Append(": ")
+							.AppendLine(action.Title);
+					}
 
-			await fixer.VerifyFixAsync(document, data.Result).ConfigureAwait(false);
+					Assert.Fail(builder.ToString());
+				}
+			}
 		}
 
 		[Test]
 		public async Task BulkUpdate([ValueSource(nameof(Samples))] SampleDataSet data)
 		{
+			if ((data.Options & SampleDataSetOptions.AllowCodeFixes) == 0)
+			{
+				return;
```

Comment:
What about `Assert.Pass`/`Assert.Skip`?
Alternatively, can we filter this out in the ValueSource?